skglm/penalties/block_separable.py

```diff
@@ -218,10 +218,13 @@ def subdiff_distance(self, w, grad, ws):
         grp_ptr, grp_indices = self.grp_ptr, self.grp_indices
 
         scores = np.zeros(len(ws))
+        grad_ptr = 0
```

Comment:
In this function, grad and w don't match: grad contains stacked gradients by group.
This is the only possible design, but it needs to be made more explicit:
- rename `grad` to `grad_ws`
- edit the docstring to reflect that gradient is restricted to ws 
- add a comment to explain that grad contains [grad_g1, grad_g2], etc.

Reply:
In hindsight this should be clarified for separable penalties too